tensorflow/lite/micro/kernels/xcore/xcore_dispatcher.cc

```diff
@@ -51,9 +52,13 @@ TfLiteStatus Dispatcher::JoinTasks() {
     // add tasks to the thread group
     size_t stack_offset = 0;
     size_t stack_words = tasks_.stack_size / kBytesPerStackword;
+
+    // align up the stack pointer (if necessary)
+    uint8_t* aligned_stack = AlignPointerUp(reinterpret_cast<uint8_t *>(tasks_.stack), kDoubleWordAlignment);
+
     for (int i = begin; i < tasks_.size; i++) {
       thread_group_add(group_, tasks_.function, tasks_.arguments[i],
-                       stack_base(&tasks_.stack[stack_offset], stack_words));
+                       stack_base(&aligned_stack[stack_offset], stack_words));
       stack_offset += tasks_.stack_size;
```

Comment:
To ensure that `stack_base` always gets an 8-byte aligned address, you may need to do something like this:
```suggestion
                       stack_base(aligned_stack, stack_words));
      aligned_stack = AlignPointerUp(&aligned_stack[tasks_.stack_size], kDoubleWordAlignment);
```
This would deprecate the `stack_offset` variable.